vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -118,7 +118,17 @@ func main() {
 	metrics_recommender.Register()
 	metrics_quality.Register()
 
-	useCheckpoints := *storage != "prometheus"
+	var usePrometheus, useCheckpoints, useNoStorage bool
+	switch *storage {
+	case "checkpoint":
+		useCheckpoints = true
+	case "prometheus":
+		usePrometheus = true
+	case "none":
+		useNoStorage = true
+	default:
+		klog.Fatalf("Storage option '%s' is not supported. Supported values: prometheus, none, checkpoint", *storage)
+	}
```

Comment:
Using bools here seems wrong - we want exactly one way to init. I'd use enum. Also I'd move as much as feasible to helper functions to keep main() small. Something like:

```
type InitSource  // this type doesn't have to be in main so let's keep it some somewhere else
const (
initFromCheckpoints initSource = iota
initFromPrometheus
noinitFromStorage
)

func getInitStorageSource(storage string) (initSource, error) { // this function doesn't have to be in main  so let's keep it some somewhere else
switch storage {
// ...
}

func InitStorage(storage string) error {  // this function doesn't have to be in main  so let's keep it some somewhere else
  initStorage, err := getInitStorageSource()
  if err != nil {
    return err
  } 
  switch initStorage {
    case initFromPrometheus:
       // ....
  }
}
```

then in main

just call `InitStorage(*storage)` and log fatal if it returns an error

Reply:
Thanks for the suggestion @jbartosik , I think that is a good way to reorganize code, true. I moved the init logic from `main` to `cluster_feeder`, please let me know what you think about this.